truffle/contracts/VaultNotary/VaultNotary.sol

```diff
@@ -0,0 +1,233 @@
+/// @author Jianwei Liu ljw725@gmail.com
+pragma solidity 0.5.0;
+
+import {Ownable} from "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+
+/** @notice The VaultNotary contract is the contract for reading/writing the
+  * vault uri and hash, and controlling the permissions to those operations by vault
+  * owner.
+  */
+contract VaultNotary is Ownable {
+    struct Data {
+        // The address of the Vault owner
+        address vaultOwner;
+        string vaultHash;
+        string vaultUri;
+
+        // The access control mapping to record whether an address can update the Uri of a Vault
+        mapping(address => bool) aclUriMapping;
+
+        // The access control mapping to record whether an address can update the Hash of a Vault
+        mapping(address => bool) aclHashMapping;
+    }
+
+    // Boolean that controls whether this contract is deprecated or not
+    bool internal isDeprecated;
+
+    // Each Vault has its own Data
+    mapping(bytes16 => VaultNotary.Data) internal notaryMapping;
+
+    // Notary Events
+    event VaultUri(address indexed msgSender, bytes16 indexed vaultId, string vaultUri);
+    event VaultHash(address indexed msgSender, bytes16 indexed vaultId, string vaultHash);
+    event VaultRegistered(address indexed msgSender, bytes16 indexed vaultId);
+
+    event UpdateHashPermissionGranted(address indexed msgSender, bytes16 indexed vaultId,
+                                                                    address indexed anotherAddress);
+
+    event UpdateHashPermissionRevoked(address indexed msgSender, bytes16 indexed vaultId,
+                                                                    address indexed anotherAddress);
+
+    event UpdateUriPermissionGranted(address indexed msgSender, bytes16 indexed vaultId,
+                                                                    address indexed anotherAddress);
+
+    event UpdateUriPermissionRevoked(address indexed msgSender, bytes16 indexed vaultId,
+                                                                    address indexed anotherAddress);
+    //better to add this event when vaultOwner can be different from msgSender, do not need currently
+    //event VaultOwnerSet(address indexed msgSender, bytes16 indexed vaultId, address indexed vaultOwner);
+
+    // Contract Events
+    event ContractDeprecatedSet(address indexed msgSender, bool isDeprecated);
+
+    /** @dev Modifier for limiting the access to vaultUri update
+      * only whitelisted user can do the decorated operation
+      */
+    modifier whitelistedOnlyForUri(bytes16 vaultId) {
```

Comment:
I think I would prefer the naming for these modifiers to be something more like `uriWhitelistedOnly` or `hashWhitelistedOnly`. Maybe even `uriAclOnly`/`uriPermissionOnly`/`uriAccessOnly`. @mlclay ?

Reply:
I'd lean towards something like `canUpdateUri`.